src/main/java/it/gov/pagopa/cgn/portal/repository/DocumentRepository.java

```diff
@@ -0,0 +1,23 @@
+package it.gov.pagopa.cgn.portal.repository;
+
+import it.gov.pagopa.cgn.portal.model.DocumentEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+public interface DocumentRepository extends JpaRepository<DocumentEntity, String> {
+
+    List<DocumentEntity> findByAgreementId(String agreementId);
+
+    @Modifying
+    @Transactional
+    @Query(value = "DELETE FROM document d " +
+            "WHERE d.agreement_k = :agreement_id "+
+            "AND cast(d.document_type as text) = :document_type", nativeQuery = true)
```

Comment:
Do we need this cast?

Reply:
This cast is required because on JPA mapping and database side we have an enum. So Hibernate fails to "cast" the value string into enum. 
On performance side, this case shouldn't be a problem because we have an index on agreement_k column of document's table.
It is not a beautiful solution, we are aware of it, but the other solutions found do not excite us more.